test/externalTests.sh

```diff
@@ -28,88 +28,238 @@
+module.exports['compilers'] = {solc: {version: "$dir/solc"} };
 EOF
+}
+
+function force_solc_settings
+{
+  local config_file="$1"
+  local settings="$2"
+  local evmVersion="$3"
+
+  printLog "Forcing solc settings..."
+  echo "Config file: $config_file"
+  echo "Optimizer settings: $settings"
+  echo "EVM version: $evmVersion"
+  echo ""
+
+  echo "module.exports['compilers']['solc']['settings'] = { optimizer: $settings, evmVersion: \"$evmVersion\" };" >> "$config_file"
+}
+
+function verify_compiler_version
+{
+  local solc_version="$1"
 
-      for optimize in "{enabled: false }" "{enabled: true }" "{enabled: true, details: { yul: true } }"
-      do
-        rm -rf build || true
-        echo "module.exports['compilers']['solc']['settings'] = {optimizer: $optimize };" >> truffle*.js
-        npx truffle compile
-        echo "Verify that the correct version ($SOLCVERSION) of the compiler was used to compile the contracts..."
-        grep -e "$SOLCVERSION" -r build/contracts > /dev/null
-        npm run test
-      done
-    )
-    rm -rf "$DIR"
+  printLog "Verify that the correct version ($solc_version) of the compiler was used to compile the contracts..."
+  grep -e "$solc_version" -r build/contracts > /dev/null
+}
+
+function clean
+{
+  rm -rf build || true
 }
 
 # Since Zeppelin 2.1.1 it supports Solidity 0.5.0.
-test_truffle Zeppelin https://github.com/OpenZeppelin/openzeppelin-solidity.git master
+printTask "Testing Zeppelin..."
+echo "==========================="
+DIR=$(mktemp -d)
+(
+  setup https://github.com/OpenZeppelin/openzeppelin-solidity.git master "$DIR"
+
+  npm install
+
+  CONFIG="truffle-config.js"
+
+  replace_libsolc_call
+  force_solc_truffle_modules "$SOLJSON"
+  force_solc "$CONFIG" "$DIR"
+
+  for optimize in "{ enabled: false }" "{ enabled: true }" "{ enabled: true, details: { yul: true } }"
+  do
+    clean
+    force_solc_settings "$CONFIG" "$optimize" "petersburg"
+
+    npx truffle compile
+    verify_compiler_version "$SOLCVERSION"
+    npm run test
+  done
+)
+rm -rf "$DIR"
+echo "Done."
+
+printTask "Testing GnosisSafe..."
+echo "==========================="
+DIR=$(mktemp -d)
+(
+  setup https://github.com/gnosis/safe-contracts.git development "$DIR"
+
+  npm install
+
+  CONFIG=$(find_truffle_config)
+
+  force_solc_truffle_modules "$SOLJSON"
+  force_solc "$CONFIG" "$DIR"
+
+  for optimize in "{ enabled: false }" "{ enabled: true }" "{ enabled: true, details: { yul: true } }"
+  do
+    clean
+    force_solc_settings "$CONFIG" "$optimize" "petersburg"
+
+    npx truffle compile
+    verify_compiler_version "$SOLCVERSION"
+    npm test
+  done
+)
+rm -rf "$DIR"
+echo "Done."
+
+printTask "Testing ColonyNetwork..."
+echo "==========================="
+DIR=$(mktemp -d)
+(
+  setup https://github.com/JoinColony/colonyNetwork.git develop "$DIR"
+
+  yarn
+  git submodule update --init
+
+  CONFIG=$(find_truffle_config)
+
+  force_solc_truffle_modules "$SOLJSON"
+  force_solc "$CONFIG" "$DIR"
+
+  cd lib
```

Comment:
This is a workaround for https://github.com/ethereum/solidity/issues/6146. I've created a PR in the 3rd party repository (https://github.com/dapphub/dappsys-monolithic/pull/2), so hopefully we can get rid of the workaround soonish.